com.unity.render-pipelines.high-definition/Editor/Lighting/LightmappingHDRP.cs

```diff
@@ -0,0 +1,156 @@
+    /// </summary>
+    public static class LightmappingHDRP
+    {
+        /// <summary>
+        ///     Bake the <paramref name="probe" /> and updated its baked texture.
+        /// </summary>
+        /// <param name="probe">The probe to bake.</param>
+        /// <param name="path">The asset path to write the baked texture to.</param>
+        /// <param name="options">The options to use for the bake.</param>
+        /// <returns>
+        ///     <c>null</c> on success or the error that occured.
+        ///     The error can be:
+        ///     * <see cref="ArgumentException" /> if the <paramref name="path" /> is invalid.
+        ///     * <see cref="ArgumentNullException" /> if the <paramref name="probe" /> is <c>null</c>.
+        ///     * <see cref="Exception" /> if the <paramref name="probe" /> is not supported. Only
+        ///     <see cref="HDAdditionalReflectionData" /> probes are currently supported.
+        /// </returns>
+        public static Exception BakeProbe([NotNull] HDProbe probe, [NotNull] string path, BakeProbeOptions options)
+        {
+            // Check arguments
+            if (probe == null || probe.Equals(null)) return new ArgumentNullException(nameof(probe));
+            if (string.IsNullOrEmpty(path))
+                return new ArgumentException($"{nameof(path)} must not be empty or null.");
+
+            // We force RGBAHalf as we don't support 11-11-10 textures (only RT)
+            var probeFormat = GraphicsFormat.R16G16B16A16_SFloat;
+            switch (probe)
+            {
+                case HDAdditionalReflectionData _:
+                {
+                    // Get the texture size from the probe
+                    var textureSize = options.textureSize.Evaluate(probe);
+
+                    // Render and write
+                    var cubeRT = HDRenderUtilities.CreateReflectionProbeRenderTarget(textureSize, probeFormat);
+                    HDBakedReflectionSystem.RenderAndWriteToFile(probe, path, cubeRT, null);
+                    cubeRT.Release();
+
+                    // Import asset at target location
+                    AssetDatabase.ImportAsset(path);
+                    HDBakedReflectionSystem.ImportAssetAt(probe, path);
+
+                    // Assign to the probe the baked texture
+                    var bakedTexture = AssetDatabase.LoadAssetAtPath<Texture>(path);
+                    probe.SetTexture(ProbeSettings.Mode.Baked, bakedTexture);
+
+                    // Mark probe as dirty
+                    EditorUtility.SetDirty(probe);
+
+                    return null;
+                }
+                case PlanarReflectionProbe _:
+                    return new Exception("Planar reflection probe baking is not supported.");
+                default: return new Exception($"Cannot handle probe type: {probe.GetType()}");
+            }
+        }
+
+        /// <summary>
+        ///     Options of <see cref="LightmappingHDRP.BakeProbe" /> function.
```

Comment:
Represents options to use with the <see cref="LightmappingHDRP.BakeProbe" /> function.